Add in integer emission curve with swop over point

SWvheerden opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Adds in a trait for use by emission curve
adds integer emission curve to consensus_manager
Adds swap over point for hot swapping across points

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
We keep getting floating point errors in testnet. This PR provides a method to swop between the two curves going forward that will keep consensus but it requires a hard_fork. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

sdbondi commented on applications/tari_app_grpc/src/generated/tari.rpc.rs:
```diff
@@ -0,0 +1,1474 @@
+/// The BlockHeader contains all the metadata for the block, including proof of work, a link to the previous block
```
This file should not be readded to the repo - was taken out in #2293

SWvheerden committed:
Adds in a trait for use by emission curve

adds integer emission curve to consensus_manager
Adds swap over point for hot swapping across points

stringhandler reviewed (approved):
Seems fine. Will need to confirm the emission before mainnet
